firedrake/paraview_reordering.py

```diff
@@ -39,13 +39,16 @@ def invert(list1, list2):
     """
     if len(list1) != len(list2):
         raise ValueError("Dimension of Paraview basis and Element basis unequal.")
-    N = len(list1)
-    pt1, o1 = zip(*sorted(zip(map(tuple, list1), range(N))))
-    pt2, o2 = zip(*sorted(zip(map(tuple, list2), range(N))))
-    if not np.allclose(pt1, pt2):
```

Comment:
Could we just adjust the tolerance in the allclose call?

Can you provide a test that exercises this?

Reply:
Adding a tolerance doesn't work because the problem occurs prior to this; also, I think allclose uses a fairly large tolerance by default. The problem is that we get the paraview nodes with a bit of floating point error in either direction (for degrees >= 5), which leads to situations like this in the sorting:
```
>>> (0.0, 0.1999999999999, 0.8) < (0.0, 0.2, 0.2)
True
>>> (0.0, 0.2, 0.8) < (0.0, 0.2, 0.2)
False
>>> sorted([(0.0, 0.1999999999999, 0.8), (0.0, 0.2, 0.2)])
[(0.0, 0.1999999999999, 0.8), (0.0, 0.2, 0.2)]
>>> sorted([(0.0, 0.2, 0.8), (0.0, 0.2, 0.2)])
[(0.0, 0.2, 0.2), (0.0, 0.2, 0.8)]
```

I will add a test sometime today.

Reply:
OK, I see, thanks.